### ISSUE TITLE
MISP Installation Error

### ISSUE THREAD

**Issue by pan01998** [MISP/MISP#9788]
### Support Questions

MISP Install Error

When I run the installation script as follows:

	wget --no-cache -O /tmp/INSTALL.sh https://raw.githubusercontent.com/MISP/MISP/2.4/INSTALL/INSTALL.sh
	bash /tmp/INSTALL.sh -c

Every time I face the below error.

	Error: Database connection "Mysql" is missing, or could not be created.
	#0 /var/www/MISP/app/Lib/cakephp/lib/Cake/Model/Datasource/DboSource.php(278): Mysql->connect()
	#1 /var/www/MISP/app/Lib/cakephp/lib/Cake/Model/ConnectionManager.php(105): DboSource->__construct()
	#2 /var/www/MISP/app/Lib/cakephp/lib/Cake/Model/Model.php(3659): ConnectionManager::getDataSource()
	#3 /var/www/MISP/app/Lib/cakephp/lib/Cake/Model/Model.php(1162): Model->setDataSource()
	#4 /var/www/MISP/app/Lib/cakephp/lib/Cake/Model/Model.php(3686): Model->setSource()
	#5 /var/www/MISP/app/Lib/cakephp/lib/Cake/Model/Model.php(1394): Model->getDataSource()
	#6 /var/www/MISP/app/Plugin/SysLogLogable/Model/Behavior/SysLogLogableBehavior.php(283): Model->schema()
	#7 /var/www/MISP/app/Lib/cakephp/lib/Cake/Model/BehaviorCollection.php(145): SysLogLogableBehavior->setup()
	#8 /var/www/MISP/app/Lib/cakephp/lib/Cake/Model/BehaviorCollection.php(66): BehaviorCollection->load()
	#9 /var/www/MISP/app/Lib/cakephp/lib/Cake/Model/Model.php(781): BehaviorCollection->init()
	#10 /var/www/MISP/app/Model/AppModel.php(113): Model->__construct()
	#11 /var/www/MISP/app/Model/User.php(239): AppModel->__construct()
	#12 [internal function]: User->__construct()
	#13 /var/www/MISP/app/Lib/cakephp/lib/Cake/Utility/ClassRegistry.php(169): ReflectionClass->newInstance()
	#14 /var/www/MISP/app/Plugin/SysLogLogable/Model/Behavior/SysLogLogableBehavior.php(279): ClassRegistry::init()
	#15 /var/www/MISP/app/Lib/cakephp/lib/Cake/Model/BehaviorCollection.php(145): SysLogLogableBehavior->setup()
	#16 /var/www/MISP/app/Lib/cakephp/lib/Cake/Model/BehaviorCollection.php(66): BehaviorCollection->load()
	#17 /var/www/MISP/app/Lib/cakephp/lib/Cake/Model/Model.php(781): BehaviorCollection->init()
	#18 /var/www/MISP/app/Model/AppModel.php(113): Model->__construct()
	#19 [internal function]: AppModel->__construct()
	#20 /var/www/MISP/app/Lib/cakephp/lib/Cake/Utility/ClassRegistry.php(169): ReflectionClass->newInstance()
	#21 /var/www/MISP/app/Lib/cakephp/lib/Cake/Console/Shell.php(316): ClassRegistry::init()
	#22 /var/www/MISP/app/Lib/cakephp/lib/Cake/Console/Shell.php(269): Shell->loadModel()
	#23 /var/www/MISP/app/Lib/cakephp/lib/Cake/Console/Shell.php(228): Shell->_loadModels()
	#24 /var/www/MISP/app/Console/Command/AppShell.php(54): Shell->initialize()
	#25 /var/www/MISP/app/Lib/cakephp/lib/Cake/Console/ShellDispatcher.php(221): AppShell->initialize()
	#26 /var/www/MISP/app/Lib/cakephp/lib/Cake/Console/ShellDispatcher.php(66): ShellDispatcher->dispatch()
	#27 /var/www/MISP/app/Console/cake.php(45): ShellDispatcher::run()
	#28 {main}
	
How can I fix the issue?

### MISP version

2.4.193

### Operating System

CentOS

### Operating System version

7

### PHP version

7.4.33

### Browser

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by UFOSmuggler** [comment]
it's having issues with your database.  check the database is up and running, and that the information within /var/www/MISP/app/Config/database.php can be used to access the database.

**Comment by pan01998** [comment]
The Installation was successful but I can't access the login page. Do I need to change something again?

**Comment by UFOSmuggler** [comment]
look in /var/www/MISP/app/tmp/logs for clues.  did you confirm that you can use the details from database.php to connect to the db?

**Comment by pan01998** [comment]
Hi
Thank you for your reply.

I found this in /var/www/MISP/app/tmp/logs/debug.log
<img width="697" alt="2" src="https://github.com/MISP/MISP/assets/173224560/f9642d86-ba91-4103-a8f9-e8889713315f">

I already confirmed the DB. That is my database.php.
<img width="419" alt="1" src="https://github.com/MISP/MISP/assets/173224560/5e14ba5b-6da6-4420-b2c4-c98ca5c406b8">
<img width="586" alt="3" src="https://github.com/MISP/MISP/assets/173224560/88596f11-4b88-4ef6-b958-6e8b89adfcb8">

**Comment by UFOSmuggler** [comment]
have you defined MISP.baseurl in the config file?

try `sudo -u www-data /var/www/MISP/app/Console/cake admin setSetting MISP.baseurl https://localhost/` or whatever your intended url is.

also i see you show mariadb running, but to see if your database.php settings are correct, try `mysql -h127.0.0.1 -umisp -p misp` although i am unsure what effect no password will have, you may need to omit the -p.  why do you have no password?

**Comment by pan01998** [comment]
Yes, I changed the baseurl. I don't know how to fix this issue, so I searched on Google and then I changed it.
I think while installing the MISP, the 'Enter password' command is prompted, so I had no idea what password I should put in, then nothing input and press the Enter key.
When I try the command you mentioned 'try MySQL -h127.0.0.1 -umisp -p misp', I don't need to press the password and I can accept the DB.

<img width="664" alt="3" src="https://github.com/MISP/MISP/assets/173224560/7df57929-5d4e-4bbb-af3e-176b24db0a38">

Thank you for your guidance.
I am really in trouble with this issue.
